### PR TITLE
protocols/noise: Introduce `NoiseAuthenticated::xx` constructor with X25519 DH key exchange

### PR DESCRIPTION
**Pull request by thomaseizinger**
# Description

This is the most common way of doing noise in libp2p and thus deserves a convenience ctor. I came across this while working on #2884, trying to reduce the noise (no pun intended) in setting up swarms for testing.

## Change checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
~- [ ] I have added tests that prove my fix is effective or that my feature works~
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Review by melekes — APPROVED** [review]
👍

File: transports/noise/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+# 0.39.1 [unreleased]
+
+- Introduce `NoiseAuthenticated::xx` constructor, assuming a X25519 DH key exchange. An XX key exchange and X25519 keys
+  are the most common way of using noise in libp2p and thus deserve a convenience constructor. See [PR XXXX].
```
Comment (mxinden) [inline transports/noise/CHANGELOG.md]:
```suggestion
  are the most common way of using noise in libp2p and thus deserve a convenience constructor. See [PR 2887].
```

File: transports/noise/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,10 @@
...
+
+[PR XXXX]: https://github.com/libp2p/rust-libp2p/pull/XXXX
```
Comment (mxinden) [inline transports/noise/CHANGELOG.md]:
```suggestion
[PR 2887]: https://github.com/libp2p/rust-libp2p/pull/2887
```

**Review by mxinden — APPROVED** [review]

**Comment by thomaseizinger** [comment]
Thanks @mxinden!

**Comment by kpp** [comment]
You forgot to add a CHANGELOG entry

**Comment by thomaseizinger** [comment]
> You forgot to add a CHANGELOG entry

There is one for `libp2p-noise`. Would you have expected a changelog entry anywhere else too?

**Comment by kpp** [comment]
I thought there should be an entry in the root [CHANGELOG.md](https://github.com/libp2p/rust-libp2p/blob/master/CHANGELOG.md)

**Comment by thomaseizinger** [comment]
> I thought there should be an entry in the root [CHANGELOG.md](https://github.com/libp2p/rust-libp2p/blob/master/CHANGELOG.md?rgh-link-date=2022-09-16T15%3A10%3A17Z)

Yep, that is true, thank you!

**Comment by thomaseizinger** [comment]
See https://github.com/libp2p/rust-libp2p/pull/2912.

**Pull request merged**